pyproject.toml

```diff
@@ -47,7 +47,7 @@ generate-setup-file = false
 [tool.poetry.dependencies]
 python = "^3.7"
 
-poetry-core = "^1.3.2"
+poetry-core = { git = "https://github.com/adriangb/poetry-core.git", branch = "allow-non-existant-path-deps" }
```

Comment:
I'm adding this temporarily as an integration test. We'll need to remove it once https://github.com/python-poetry/poetry-core/pull/507#pullrequestreview-1169630432 gets merged and a new release is made. The good news is that this PR isn't adding any functionality, just tests, so in theory just by merging https://github.com/python-poetry/poetry-core/pull/507#pullrequestreview-1169630432 and releasing all of this will work for users -> we don't need to release in lockstep or anything like that.